### PR TITLE
[106][135] refactor import and export checkfiles and minor bug fix on health check dashboard

### PR DESCRIPTION
**Pull request by connectbo**
This pr refactored how import and export of check-files work. By using `require.context` in webpack, ZMT will be able to load all javascript objects in that directory (./custom_checkfiles and ./default_checkfiles) into an array. Also removed double quotes in js objects.

### CODE REVIEW

**Comment by trel** [comment]
awesome and very elegant.

without running it myself yet...  does it handle 'bad' JSON cleanly?

if it fails an import - can it gracefully tell the admin where the syntax error is?

does it still load the others that are good?

**Comment by connectbo** [comment]
hold off on it actually...  found something weird, check status filtering is not working :)

**Comment by connectbo** [comment]
alright, error fixed - should be working. I guess if we want to display syntax warnings gracefully to the admin, I can set eslint rules to be less strict. But if that's something that react can't compile, the error message might still appear in the browser.

**Comment by trel** [comment]
Can we 'try' to lint them first, and only if they pass/load cleanly... then we actually use them?

if they don't load cleanly, we can catch the error, populate something onscreen (or in a data structure), and then keep looping through the rest of them?

**Comment by trel** [comment]
Looking pretty good.

1) I was able to create a custom checkfile with empty name and description and it loaded.    i guess validating these checkfiles directly will be a part of getting https://github.com/irods/irods_client_zone_management_tool/issues/105 implemented.

2) Saw that editing the interval didn't enforce an integer on the 'seconds' (I entered '5f').
I see 'NaN' in the column on the right...

Screen Shot 2022-03-02 at 1 42 34 PM

3) Let's update the template to 'work' out of the box...

```diff
$ git diff
diff --git a/src/data/checkfiles/checkfile_template.js b/src/data/checkfiles/checkfile_template.js
index 2c25fd4..70f4da8 100644
--- a/src/data/checkfiles/checkfile_template.js
+++ b/src/data/checkfiles/checkfile_template.js
@@ -10,7 +10,18 @@ export const checkfile_template = {
         //   status: '<ok,unexpected,warning,bad>',
         //   message: 'message that will be displayed on homepage'
         // }
-    
-        // your code here
+
+       // default result object
+        let result = {
+            status: 'ok',
+            message: '',
+            success: 0,
+            failed: []
+        }
+
+       // your code here
+
+       // return
+        return result
     }
-}
\ No newline at end of file
+}
```

**Comment by trel** [comment]
this also will close #104

**Comment by connectbo** [comment]
alright. Drafted updates for README, and made a validate function when loading each checkfile (non-empty string for name, description, a positive integer for intervals, and checker function should be defined). And the input for interval is now positive integer only.

Feel free to suggest if any, and can discuss more about #105.

**Comment by trel** [comment]
I see some weirdness / inconsistency with the minimum_server_version vs. min_server_version... let's use all the letters...

And it still didn't pick up my 'interval_in_seconds` correctly...

Screen Shot 2022-03-03 at 9 24 35 PM

And it looks like I should be using 'healthy' rather than 'ok'...

Screen Shot 2022-03-03 at 10 05 08 PM

File: src/data/checkfiles/checkfile_template.js [hunk 1]
```diff
@@ -3,14 +3,19 @@ export const checkfile_template = {
...
-    checker: function() {
-        // @returns{object} {
-        //   status: '<ok,unexpected,warning,bad>',
-        //   message: 'message that will be displayed on homepage'
-        // }
-    
+    checker: function () {
+        // default result object
+        let result = {
+            status: 'ok',
```
Comment (trel) [inline src/data/checkfiles/checkfile_template.js:11]: 'ok' -> 'healthy'

Comment (connectbo) [reply src/data/checkfiles/checkfile_template.js:11]: oops, that's true

**Comment by connectbo** [comment]
Interesting. What is your value for `interval_in_seconds`? I think I'm letting ZMT check if that's defined and a positive value.

**Comment by trel** [comment]
> Interesting. What is your value for `interval_in_seconds`? I think I'm letting ZMT check if that's defined and a positive value.

just 15

so yes, please try and break it some more :)

File: src/data/checkfiles/default_checkfiles/bad-checkfiles.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export default {
```
[the remark this answers is not in the document]
Comment (connectbo) [reply src/data/checkfiles/default_checkfiles/bad-checkfiles.js:1]: lol I was for sure not functioning yesterday :| will remove

File: src/data/checkfiles/default_checkfiles/rest-api-auth-endpoint.js [hunk 3]
```diff
@@ -3,8 +3,8 @@ import axios from 'axios'
 export const restApiAuthEndpoint = {
```
[the remark this answers is not in the document]
Comment (connectbo) [reply src/data/checkfiles/default_checkfiles/rest-api-auth-endpoint.js:3]: oops I forgot to change this.

**Comment by connectbo** [comment]
Just pushed some changes. Is there anything in the README that I should revise a little bit before I squash the commits?

**Comment by trel** [comment]
I can tweak the README later for readability.  It covers the breadth of how it works, that's the most important part this week.

Please squash down to good commits, make sure you point to 104 as well.   And we'll get them #'d after review.

**Comment by connectbo** [comment]
yep, squashed down to four commits. does that look good?

**Comment by trel** [comment]
Well, let's remove "[105]" in the last commit message, since we're not defining the JSON Schema, just spot-enforcing what we want via the code itself.

Remove that, and please # em.

**Comment by connectbo** [comment]
yep removed and added #

**Pull request merged**